static/app/components/profiling/flamegraph/flamegraphSpans.tsx

```diff
@@ -367,6 +392,108 @@ export function FlamegraphSpans({
+    if (!value) {
+      addErrorMessage(t('Event description value is empty.'));
+      return;
+    }
+
+    navigator.clipboard
+      .writeText(value)
+      .then(() => {
+        addSuccessMessage(t('Description copied to clipboard'));
+      })
+      .catch(() => {
+        addErrorMessage(t('Failed to copy description to clipboard'));
+      });
+  }, [hoveredNodeOnContextMenuOpen]);
+
+  const onCopyOperation = useCallback(() => {
+    const value = hoveredNodeOnContextMenuOpen.current?.node.span.op ?? '';
+    if (!value) {
+      addErrorMessage(t('Event operation value is empty.'));
+      return;
+    }
+
+    navigator.clipboard
+      .writeText(value)
+      .then(() => {
+        addSuccessMessage(t('Operation copied to clipboard'));
+      })
+      .catch(() => {
+        addErrorMessage(t('Failed to copy operation to clipboard'));
+      });
+  }, [hoveredNodeOnContextMenuOpen]);
+
+  const onCopyEventId = useCallback(() => {
+    const value =
+      hoveredNodeOnContextMenuOpen.current?.node.span.event_id ??
+      hoveredNodeOnContextMenuOpen.current?.node.span.span_id ??
+      '';
+    if (!value) {
+      addErrorMessage(t('Event ID value is empty.'));
+      return;
+    }
+
+    navigator.clipboard
+      .writeText(value)
+      .then(() => {
+        addSuccessMessage(t('Event ID copied to clipboard'));
+      })
+      .catch(() => {
+        addErrorMessage(t('Failed to copy event ID to clipboard'));
+      });
+  }, [hoveredNodeOnContextMenuOpen]);
+
+  const onOpenInTraceView = useCallback(() => {
+    const node = hoveredNodeOnContextMenuOpen.current?.node;
+    if (!node) {
+      addErrorMessage(t('No event ID or span ID found'));
+      return;
+    }
+
+    const nodePath: Record<string, string> = {};
+
+    if (node.span.op === 'transaction' && node.span.event_id) {
+      // If the user clicks on a transaction, we can directly use the event_id
+      nodePath.eventId = node.span.event_id;
+    } else if (node.span.span_id) {
+      // If the user clicks on a span, we need to traverse up the tree to find the transaction so that
+      // the trace view knows which transaction to load and what span to point to.
+      nodePath.spanId = node.span.span_id;
+      let parent = node.parent;
+      while (parent) {
+        if (parent.span.op === 'transaction' && parent.span.event_id) {
+          nodePath.eventId = parent.span.event_id;
+          break;
+        }
+        parent = parent.parent;
+      }
+    }
+
+    const link = getTraceDetailsUrl({
+      traceSlug: node.span.trace_id,
+      dateSelection: normalizeDateTimeParams(pageFilters.selection.datetime),
+      location,
+      organization,
+      timestamp: node.span.timestamp,
+      source: TraceViewSources.PROFILING_FLAMEGRAPH,
+      ...nodePath,
+    });
+
+    window.open(
+      `${link.pathname}?${qs.stringify(link.query ?? {})}`,
+      '_blank',
+      'noopener,noreferrer'
+    );
```

Comment:
Should this use `const navigate = useNavigate()`?

Reply:
If I understood it correctly the reason you have to normalize the URL below is because you're not using the `navigate` or `browserHistory` wrappers.

Reply:
Hmm, let me see, but idk if there is a way to useNavigate to a separate tab

Reply:
Apparently, useNavigate cannot open in a new tab.. I would keep this as is in this case, unless we want to open in the same tab, which I'm not sure we do tbqh